src/components/layers/layers/AddLayerButton.js

```diff
@@ -0,0 +1,35 @@
+import React, { Fragment, useState } from 'react';
+import i18n from '@dhis2/d2-i18n';
+import MenuButton from '../../core/MenuButton';
+import AddIcon from '@material-ui/icons/AddCircleOutline';
+import AddLayerPopover from './AddLayerPopover';
+import styles from './styles/AddLayerButton.module.css';
+
+const AddLayerButton = () => {
+    const [isOpen, setIsOpen] = useState(false);
+    const [buttonRef, setButtonRef] = useState();
+    const toggleDialog = () => setIsOpen(!isOpen);
+
+    return (
+        <Fragment>
+            <div className={styles.addLayerBtn} ref={setButtonRef}>
```

Comment:
I'm not sure about this?
Shouldn't it just use the `useRef` hook?

Reply:
Switched to useRef hook.